api/src/main/java/org/eclipse/microprofile/config/spi/ConfigValueSource.java

```diff
@@ -0,0 +1,185 @@
+ * <p>
+ *
+ * A <em>configuration source</em> is always read-only; any potential updates of the backing configuration values must
+ * be handled directly inside each configuration source instance.
+ *
+ * <h3 id="default_config_sources">Default configuration sources</h3>
+ *
+ * Some configuration sources are known as <em>default configuration sources</em>.  These configuration sources are
+ * normally available in all automatically-created configurations, and can be
+ * {@linkplain ConfigBuilder#addDefaultSources() manually added} to manually-created configurations as well. The
+ * default configuration sources are:
+ *
+ * <ol>
+ *   <li>{@linkplain System#getProperties() System properties}, with an {@linkplain #getOrdinal() ordinal value} of
+ *   {@code 400}</li>
+ *   <li>{@linkplain System#getenv() Environment properties}, with an ordinal value of {@code 300}</li>
+ *   <li>The {@code /META-INF/microprofile-config.properties} {@linkplain ClassLoader#getResource(String) resource },
+ *   with an ordinal value of {@code 100}</li>
+ * </ol>
+ *
+ * <h3>Environment variable name mapping rules</h3>
+ *
+ * Some operating systems allow only alphabetic characters or an underscore ({@code _}) in environment variable
+ * names. Other characters such as {@code .}, {@code /}, etc. may be disallowed. In order to set a value for a config
+ * property that has a name containing such disallowed characters from an environment variable, the following rules are
+ * used.
+ *
+ * Three environment variables are searched for a given property name (e.g. "{@code com.ACME.size}"):
+ * <ol>
+ *   <li>The exact name (i.e. "{@code com.ACME.size}")</li>
+ *   <li>The name, with each character that is neither alphanumeric nor _ replaced with _ (i.e. "{@code com_ACME_size
+ *   }")</li>
+ *   <li>The name, with each character that is neither alphanumeric nor _ replaced with _ and then converted to upper
+ *   case (i.e. "{@code COM_ACME_SIZE}")</li>
+ * </ol>
+ *
+ * <p>
+ * The first of these environment variables that is found for a given name is returned.
+ *
+ * <h3 id="discovery">Configuration source discovery</h3>
+ *
+ * Discovered configuration sources are loaded via the {@link java.util.ServiceLoader} mechanism and and can be
+ * registered by providing a resource named {@code META-INF/services/org.eclipse.microprofile.config.spi.ConfigSource},
+ * which contains the fully qualified {@code ConfigSource} implementation class name as its content.
+ * <p>
+ *
+ * Configuration sources may also be added by defining {@link org.eclipse.microprofile.config.spi.ConfigSourceProvider}
+ * classes which are discoverable in this manner.
+ *
+ * <h3>Closing configuration sources</h3>
+ *
+ * If a configuration source implements the {@link AutoCloseable} interface, then its {@linkplain AutoCloseable#close()
+ * close method} will be called when the underlying configuration is released.
+ *
+ * @author <a href="mailto:radcortez@yahoo.com">Roberto Cortez</a>
```

Comment:
For copyright and IP issue, I think you need to lift all contributors from ConfigSource file here.

Reply:
Did you consult with a lawyer?

Reply:
I had no idea what to do there. I'm fine either way.